[feat] #36 header button을 url에 따라 조정

sewonkimm opened the pull request:
props에 path 값을 전달해 보내서 url에 맞는 버튼만 보이도록 렌더링

sewonkimm committed:
[feat] #36 header button을 url에 따라 조정

props에 path 값을 전달해 보내서 url에 맞는 버튼만 보이도록 렌더링

sewonkimm committed:
Merge branch 'develop' of https://github.com/mojimojicorp/MojimojiMemoji into feat/#36/header

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -1,18 +1,45 @@
-const Header = () => {
+export interface Props {
+  path: string;
+}
```
Props는 다른 곳에서 사용하지 않으니 export 하지 않아도 될 것 같습니다!

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -1,18 +1,45 @@
+  const [select, setSelect] = useState(false);
+  const [logout, setLogout] = useState(false);
+  const [save, setSave] = useState(false);
+  const [back, setBack] = useState(false);
```
type이 true일 경우에는 변수명을 is 나 has 등으로 시작하는 것이 좋을 것 같아요! 
참고글 : https://soojin.ro/blog/naming-boolean-variables 등등

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -1,18 +1,45 @@
+      {select ? <div className={styles.select_btn} /> : null}
+      {save ? <div className={styles.save_btn} /> : null}
+      {logout ? <div className={styles.logout_btn} /> : null}
+      {back ? <div className={styles.back_btn} /> : null}
```
`{ back && <div className={styles.back_btn} />}`
이런식이 가독성이 더 높아보이는데 어떠신가요??

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -1,18 +1,45 @@
+      setSave(true);
+      setBack(true);
+    }
+  });
```
useEffect는 두번째인자로 배열을 꼭 넣어주는 것이 좋습니다 : )

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -1,18 +1,45 @@
+  const [back, setBack] = useState(false);
+
+  useEffect(() => {
+    if (path == 'folder-list') {
```
비교는 === 으로ㅎㅎ 오타인 것 같아용

sewonkimm committed:
[Refact] #36 header button loading

1. 화면이 2번 렌더링 되는 것을 줄이기 위해 useEffect 제거
2. 가독성을 향상하기 위해 state 사용을 제거하고 props로 값을 넘겨주는
방식으로 구조를 변경

sewonkimm committed:
[Chore] #38 웹 표준, scss mixin 적용

웹 표준 가이드에 따라 div 태그를 header, button등으로 명시해주고,
scss에서 중첩되는 내용들은 mixin을 사용해 가독성을 높임.

JaeHyeonKim19 reviewed (approved):
수고하셨습니다.

sewonkimm committed:
[Feat] #38 header 버튼 기능 구현

1. classnames 설치
2. FolderListPage, MemoListPage에 있던 FolderSetting, MemoSetting 컴포넌트를 Header에 포함시키는 구조로 변경
3. select버튼 선택시 이미지 변경과 페이지에 맞는 setting가 보일 수 있도록 toggle 구현

soyoungjeong commented on app/src/components/Header/index.tsx:
```diff
@@ -14,17 +18,36 @@ const Header = ({
   hasSaveBtn,
   hasBackBtn,
 }: HeaderProps) => {
+  const toggleSelect = () => {
```
이름이 동사+동사라 좀 어색한듯 하기도 하네요ㅎㅎ toggleSelectBtn 이나 handleSelectBtnClick 등은 어떨까요??

sewonkimm replied:
앗 그렇네요! 동사+명사로 수정해보겠습니다 👍

sewonkimm committed:
[Chore] #38 function 이름 수정

함수 기능을 쉽게 알아보게 하기 위해 함수명을 '명사+명사'에서 '동사+명사'로 변경

soyoungjeong reviewed (approved):
수고하셨습니다 :)
